### PR TITLE
Example of u8glib draw loop

### PR DESCRIPTION
**Pull request by UncleRedz**
Example of an u8glib draw loop that works with WiFi by keeping the draw
loop execution time to a minimum. This was mentioned in issue #1178 as something that is missing in the existing u8glib lua examples which focus on demoing the u8glib functionality and not necessarily showing how the draw loop should be implemented in a nice way.

### CODE REVIEW

**Comment by TerryE** [comment]
@UncleRedz I was just looking up your code API ref and I noticed a typo.  Can you add an update to `## u8g.disp:nextPage()` line  `## u8g.disp:nextPage()`+1 is missing a `]` after the `[u8glib nextPage()` ?

I also had a play with your example  -- at a compile level only since I don't have a u8g compatible display -- and came across one piece of Lua syntactic subtlety that I hadn't twigged before.  `local function drawPages()` is _not_ the same as  `local drawPages = function ()` as the former looks the local slot `drawPages` before starting the compile, so any recursive references correctly self-reference the function, but the latter compiles the function then books the local so the local doesn't exists during the function compile and any recursive references actually reference a (non-existent) global function.  Have a play with `luac.cross -l` to see what I mean.  

Another good hint is to search the listing for `GLOBAL` references and make sure that you are only using globals when you intend to.

You can also nested functions and use upvalues from the outer function within the inner function to create persistent variables.  I've create my cut down version of your u8g_drawloop.lua in [this gist](https://gist.github.com/TerryE/6c06c5f66496bea8a58b) which you might want to look at for some ideas / Lua techniques, and feel free to pinch bits if you like them.

It's also nice to see node.task.post being put to good use! :smile:  

Thanks for this, Terry

File: lua_examples/u8glib/u8g_drawloop.lua [hunk 1]
```diff
@@ -0,0 +1,75 @@
+------------------------------------------------------------------------------
+-- u8glib example which shows how to implement the draw loop without causing
+-- timeout issues with the WiFi stack. This is done by drawing one page at
+-- a time, allowing the ESP SDK to do it's house keeping between the page
```
Comment (marcelstoer) [inline lua_examples/u8glib/u8g_drawloop.lua]: It's 'its' rather than 'it's' :wink:

**Comment by UncleRedz** [comment]
@TerryE, that's an interesting observation of the local function creation, I'll keep that in mind. I've looked at your gist and incorporated the parts which I think makes sense for this example. The draw loop is now more tightly contained / encapsulated.

As I'm new to git and github, what's the correct procedure from here? Squash the commits and rebase as described in the contributor documentation? It doesn't state how to handle updates to your PR.

**Comment by marcelstoer** [comment]
> Squash the commits and rebase as described in the contributor documentation?

Yes, that's the correct.

**Comment by UncleRedz** [comment]
OK, I think I've squashed and rebased now, please let me know if I there are any issues.

**Comment by marcelstoer** [comment]
There are...you have six commits instead of one. I'll still merge this, though.

**Comment by devsaurus** [comment]
@UncleRedz thanks a lot for showing how to benefit from the new task interface.
I'll update the remaining u8glib examples accordingly.

**Pull request merged**